### ISSUE TITLE
Custom setup

### ISSUE THREAD

**Issue by stefjoosten** [ndmitchell/ghcid#299]
In the file `package.yaml` I can configure a custom setup. As a result, the compiler runs setup.hs. I reckon ghcid should do the same to ensure that changes in setup.hs are also seen by ghcid.

**Comment by ndmitchell** [comment]
Are you suggesting that changes to Setup.hs cause ghcid to restart, much like changes to foo.cabal do? The code is at https://github.com/ndmitchell/ghcid/blob/master/src/Ghcid.hs#L134-L162, and I'd welcome a patch (it's the things that end up in `restart`. I also note that `package.yaml` does not trigger a restart - that sounds like a bug?

**Comment by stefjoosten** [comment]
No, the problem is that stack install behaves differently than ghcid. Stack install runs the program setup.hs if package.yaml specifies a custom setup. And ghcid doesn't.

**Comment by ndmitchell** [comment]
What is the command that ghcid is running? It should say on the first few lines of the output.

**Comment by tim2CF** [comment]
Having similar issue, seems like `ghcid` command runs old version of `Setup.hs` after the changes in setup file, even in case where `--restart=Setup.hs` is specified.
